apps/meteor/client/views/omnichannel/customFields/EditCustomFields.tsx

```diff
@@ -0,0 +1,226 @@
+} from '../../../components/Contextualbar';
+import { useFormsSubscription } from '../additionalForms';
+import { useRemoveCustomField } from './useRemoveCustomField';
+
+const getInitialValues = (customFieldData: Serialized<ILivechatCustomField> | undefined) => ({
+	field: customFieldData?._id || '',
+	label: customFieldData?.label || '',
+	scope: customFieldData?.scope || 'visitor',
+	visibility: customFieldData?.visibility === 'visible',
+	searchable: !!customFieldData?.searchable,
+	regexp: customFieldData?.regexp || '',
+	// additional props
+	type: customFieldData?.type || 'input',
+	required: !!customFieldData?.required,
+	defaultValue: customFieldData?.defaultValue || '',
+	options: customFieldData?.options || '',
+	public: !!customFieldData?.public,
+});
+
+const EditCustomFields = ({ customFieldData }: { customFieldData?: Serialized<ILivechatCustomField> }) => {
+	const t = useTranslation();
+	const router = useRouter();
+	const queryClient = useQueryClient();
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const { useCustomFieldsAdditionalForm } = useFormsSubscription();
+	const AdditionalForm = useCustomFieldsAdditionalForm();
+
+	const handleDelete = useRemoveCustomField();
+
+	const methods = useForm({ mode: 'onBlur', values: getInitialValues(customFieldData) });
+	const {
+		control,
+		handleSubmit,
+		formState: { isDirty, errors },
+	} = methods;
+
+	const saveCustomField = useMethod('livechat:saveCustomField');
+
+	const handleSave = useMutableCallback(async ({ visibility, ...data }) => {
+		try {
+			await saveCustomField(customFieldData?._id as unknown as string, {
+				visibility: visibility ? 'visible' : 'hidden',
+				...data,
+			});
+
+			dispatchToastMessage({ type: 'success', message: t('Saved') });
+			queryClient.invalidateQueries(['livechat-customFields']);
+			router.navigate('/omnichannel/customfields');
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	});
+
+	const scopeOptions: SelectOption[] = useMemo(
+		() => [
+			['visitor', t('Visitor')],
+			['room', t('Room')],
+		],
+		[t],
+	);
+
+	const formId = useUniqueId();
+	const fieldField = useUniqueId();
+	const labelField = useUniqueId();
+	const scopeField = useUniqueId();
+	const visibilityField = useUniqueId();
+	const searchableField = useUniqueId();
+	const regexpField = useUniqueId();
+
+	return (
+		<Contextualbar>
+			<ContextualbarHeader>
+				<ContextualbarTitle>{customFieldData?._id ? t('Edit_Custom_Field') : t('New_Custom_Field')}</ContextualbarTitle>
+				<ContextualbarClose onClick={() => router.navigate('/omnichannel/customfields')} />
+			</ContextualbarHeader>
+			<ContextualbarScrollableContent>
+				<FormProvider {...methods}>
+					<form id={formId} onSubmit={handleSubmit(handleSave)}>
+						<FieldGroup>
+							<Field>
+								<FieldLabel htmlFor={fieldField} required>
+									{t('Field')}
+								</FieldLabel>
+								<FieldRow>
+									<Controller
+										name='field'
```

Comment:
Let's add some validations to the FE for this field, in the BE we check for some things 

![Screen Shot 2023-10-27 at 15 59 15](https://github.com/RocketChat/Rocket.Chat/assets/20868078/b0216922-e3c2-4a38-94a6-20dcbd3a1acd)